### PR TITLE
Move include/event.php to src/

### PR DESCRIPTION
**Pull request by MrPetovan**
Part of #3878

### CODE REVIEW

File: src/Content/Widget/CalendarExport.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+//
+//		/*
+//		 * If it's a kind of profile page (intval($owner_uid)) return if the user not logged in and
+//		 * export feature isn't enabled.
+//		 */
+//		if (intval($owner_uid) && ! local_user() && ! Feature::isEnabled($owner_uid, "export_calendar")) {
+//			return;
+//		}
+		/*
+		 * All the legacy checks above seem to be equivalent to the check below, see https://ethercalc.org/z6ehv1tut9cm
```
Comment (MrPetovan) [inline src/Content/Widget/CalendarExport.php]: In your review please pay special attention to this, I've created a spreadsheet to simplify the permission condition and the result is a little bit surprising: https://ethercalc.org/z6ehv1tut9cm

File: src/Content/Widget/CalendarExport.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		 * If there is a mistake in the spreadsheet, please notify @MrPetovan on GitHub or by email mrpetovan@gmail.com
+		 */
+		if (!local_user()) {
```
Comment (rabuzarus) [inline src/Content/Widget/CalendarExport.php]: This would mean that visitors will not be able to export the events in the `/ cal 'module, but they should be able to export if the profile owner has enabled the' export_calendar 'option

Comment (MrPetovan) [reply src/Content/Widget/CalendarExport.php]: This is what the three permissions checks amount to after careful study, see comment above. It probably isn't intended, but it's the only logical outcome from the code.

Comment (rabuzarus) [reply src/Content/Widget/CalendarExport.php]: You have a mistake in your spreadsheet in line 3 (TRUE, FALSE, FALSE, TRUE). The third check passes because the Feature is enabled

Comment (MrPetovan) [reply src/Content/Widget/CalendarExport.php]: Ah, thanks!

**Comment by MrPetovan** [comment]
Thanks @rabuzarus, it is fixed.

File: src/Content/Widget/CalendarExport.php [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		// It's not the private events page (we don't get the $owner_uid for /events).
+		if (!local_user() && !$owner_uid) {
+			return;
+		}
+
+		/*
+		 * If it's a kind of profile page (intval($owner_uid)) return if the user not logged in and
+		 * export feature isn't enabled.
+		 */
+		if (!local_user() && $owner_uid && !Feature::isEnabled($owner_uid, 'export_calendar')) {
```
Comment (rabuzarus) [inline src/Content/Widget/CalendarExport.php]: Why did you remove the second check? Think of a `local_user()` who visit a foreign profile on the same server. Without the second check the `local_user()` could export the events of a foreign profile although the profile owner has disabled the calendar export

Comment (MrPetovan) [reply src/Content/Widget/CalendarExport.php]: According to the chart, the second check doesn't matter, it passes the same way as the first one.

Comment (rabuzarus) [reply src/Content/Widget/CalendarExport.php]: Please be clearer. I don't get it.

The first check ensures that the widget is only shown if it is a local system page (like `events`) or a profile page (like `cal` -- note: `$owner_uid` is only available on profile pages)

The second check ensures that the `local_user()` does access to the widget on foreign profile pages on the same server if the profile owner has disabled the widget. At the same time it ensures that the `local_user()` does have access to the export widget if it is his/her own profile page (`/cal` module).

The third check is the regular check which checks if the feature is enabled for visitors.

What do you mean with:
> For example I'm surprised we don't test the feature on its own, which would actually prevent the widget to show if the user disabled the feature.

The widget should appear for the local users `/events` and his/her own `/cal` page, regardless of whether it is enabled or not

Comment (MrPetovan) [reply src/Content/Widget/CalendarExport.php]: What I mean is that the original set of permissions checks probably weren't exactly doing what was expected from them, unless I messed up again in my spreadsheet, of course.

I believe my current version is logically equivalent to the original thanks to your help, but it might not be "right" as "not doing what was expected" but it would have been true of the original version as well.

I don't have a problem fixing whatever needs fixing if we can come up with a complete and intelligible set of permissions checks.

Comment (MrPetovan) [reply src/Content/Widget/CalendarExport.php]: It turns out I was wrong again, I decided to trust the machine rather than myself, and wrote formulas in my spreadsheet. It turns out I had a couple of mistakes in the 2nd check, which makes it actually cover the single case the 3rd check did.

I ended up keeping only two checks, and I verified it keeps the same truth table as before in the spreadsheet.

File: src/Content/Widget/CalendarExport.php [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+		 */
+		/*
+		 * Cal logged in user (test permission at foreign profile page).
+		 * If the $owner uid is available we know it is part of one of the profile pages (like /cal).
+		 * So we have to test if if it's the own profile page of the logged in user
+		 * or a foreign one. For foreign profile pages we need to check if the feature
+		 * for exporting the cal is enabled (otherwise the widget would appear for logged in users
+		 * on foreigen profile pages even if the widget is disabled).
+		 */
+		if (local_user() != $owner_uid && !Feature::isEnabled($owner_uid, "export_calendar")) {
```
Comment (rabuzarus) [inline src/Content/Widget/CalendarExport.php]: This check doesn't respect the case where `local_user()` didn't enabled the the `export_calendar` widget and visits his/her `/events` module. With your checks the widget would not appear now because `$owner_uid` would be `null`.

- `local_user()` should be able to see the widget for `/events` and his/her own `/cal` page (even if `export_calendar` is disabled)
- if `local_user()` is visiting a foreign profile page we have to respect if `export_calendar` is enabled (the situation on the local friendica instance)
- if a normal visitor is visiting a profile page we have to respect if `export_calendar` is enabled (the situation on foreign friendica instances)

Note: there seems to be a bug in the `/mod/events.php`. According to the code the widget should be loaded if `$a->argc > 1` but I think it should be loaded if `$a->argc >= 1`

Comment (MrPetovan) [reply src/Content/Widget/CalendarExport.php]: How do you access your own `/events` and `/cal` page? I had to manually manipulate the URL to reach the `/cal/[username]` page, and I still am not sure about the `/events` page. Additionally, with my new spreadsheet, I'm now sure this case wasn't covered before.

> Note: there seems to be a bug in the /mod/events.php. According to the code the widget should be loaded if `$a->argc > 1` but I think it should be loaded if `$a->argc >= 1`

`$a->argc` contains at least the module name (here, `events`), so the condition is correct.

Thanks for bearing with me!

Comment (MrPetovan) [reply src/Content/Widget/CalendarExport.php]: Can @annando answer while he's lurking around?

Comment (annando) [reply src/Content/Widget/CalendarExport.php]: Until yet I haven't watched this issue, since @rabuzarus seems to be active on this.

Comment (MrPetovan) [reply src/Content/Widget/CalendarExport.php]: I just need an answer to `How do you access your own /events and /cal page?`

Comment (tobiasd) [reply src/Content/Widget/CalendarExport.php]: What is the problem with accessing these URLs? `/events` is in the navbar of the theme and `/cal` is for visitors of the profile. Or am I mistaken?

Comment (MrPetovan) [reply src/Content/Widget/CalendarExport.php]: - `/cal/[username]` can be accessed on someone else's profile but not on yourself. Widget is not shown. `$owner_uid` is set.
- `/events` is your personal calendar with the events you can see. Widget is not shown.
- `/events/[something]` is supposed to show the Widget, but I have no idea what `[something]` is supposed to be nor how to access it from the interface.

Comment (annando) [reply src/Content/Widget/CalendarExport.php]: AFAIK ```/events``` is only visible for yourself. ```/cal``` should be visible for everyone for public events and events that can be seen by the remote user.

Comment (MrPetovan) [reply src/Content/Widget/CalendarExport.php]: So it doesn't matter if `$owner_uid` is set or not, and the checks are correct.

Comment (annando) [reply src/Content/Widget/CalendarExport.php]: I don't know what happens when you visit ```/cal``` as a stranger and you click on a event. There could be called some ```/events``` folder.

Comment (MrPetovan) [reply src/Content/Widget/CalendarExport.php]: When a stranger tries to access my `/cal/hypolite` page, the query takes 517 seconds to complete, so I really can't tell you.

Comment (rabuzarus) [reply src/Content/Widget/CalendarExport.php]: Arghh, again. Lets say you are `local_user()` (let's say with `local_user() = 4) and you have disabled the widget (so value is `false`) than this check would lead to a return for `/events`.
`if (4 != null && false) {return;}`

BUT:
The `local_user()` should always see his/her own export widget on his/her own pages (`/events` and `/cal`) regardless of whether  the function has been enabled or not. The setting should be only for allowing visitors to see the widget or not.

With your current checks the `local_user()` can't see his/her own export widget if he/she has disabled it

Comment (MrPetovan) [reply src/Content/Widget/CalendarExport.php]: But the widget code isn't even called when `/events` is accessed without subsequent URI parameters, so the check isn't even evaluated in the first place.

On the other hand, while visiting `/cal/[own username]` the  `$owner_uid` is set and the widget correctly shows for self even with the feature disabled.

Comment (rabuzarus) [reply src/Content/Widget/CalendarExport.php]: > But the widget code has never been called when /events is accessed without subsequent URI parameters

yes this is a bug in `/mod/events.php`. Since `/cal` isn't easy accessible for the `local_user()`, the `local_user()` should have the possibility to export his/her own events under `/events`

Comment (MrPetovan) [reply src/Content/Widget/CalendarExport.php]: Finally, we got somewhere!

Comment (MrPetovan) [reply src/Content/Widget/CalendarExport.php]: Done.

**Comment by MrPetovan** [comment]
Can this be merged please?

**Comment by annando** [comment]
I would like to delegate the approval to @rabuzarus - he watched this PR more than me.

**Comment by rabuzarus** [comment]
I wait for an answer for https://github.com/friendica/friendica/pull/4622#discussion_r175368096

This condition would lead to a return if the feature is disabled and the `local_user()` visits his/her `/events` page

**Comment by MrPetovan** [comment]
It's been addressed in https://github.com/friendica/friendica/pull/4622/commits/6c448dff63a11ffc8c230a10b2b9375451c22896

Instead of making the Widget condition more complicated, I set the `$owner_uid` in the `/events` page as well.

**Comment by rabuzarus** [comment]
ahhh ok, I was wrong. @MrPetovan changed the `/mod/events.php` so `$a->data[`user]` is available for `/events`

@annado Are there any internal (unknown) specs when `$a->data['user']` should be available. At the moment we use it only in modules which are also accessible for the public

**Comment by annando** [comment]
I haven't had a deep look at this data structure.

**Comment by MrPetovan** [comment]
This has been rebased against the RC changes, please merge #4651 first
